system-addon/content-src/components/Card/Card.jsx

```diff
@@ -131,11 +194,24 @@ export class _Card extends React.PureComponent {
     this.maybeLoadImage();
   }
 
+  componentWillMount() {
+    this.updateCardImage(this.props);
+  }
+
   componentWillReceiveProps(nextProps) {
     // Clear the image state if changing images
-    if (nextProps.link.image !== this.props.link.image) {
+    if (nextProps.link && !this.isImageInState(nextProps.link.image)) {
       this.setState({imageLoaded: false});
     }
+
+    // NB: updateCardImage needs to be called after isImageInState since cardImage's
+    // state is modified in updateCardImage and isImageInState uses the old state.
+    this.updateCardImage(nextProps);
```

Comment:
I've read through this comment a few times and am still confused. What's the ordering of events / lifecycle stages that leads to this needing to be here?

Reply:
We are calling `updateCardImage` in `componentWillMount` during the mount process. For the update process, we cannot call `updateCardImage` in `componentWillUpdate` because `updateCardImage` calls `this.setState`, and hence we're calling it in `componentWillReceiveProps`.

From `componentWillUpdate() / UNSAFE_componentWillUpdate()`:
> Note that you cannot call this.setState() here; nor should you do anything else (e.g. dispatch a Redux action) that would trigger an update to a React component before UNSAFE_componentWillUpdate() returns.
>
> If you need to update state in response to props changes, use getDerivedStateFromProps() instead.
> See: https://reactjs.org/docs/react-component.html#updating

I have updated this part of the code to support React >= 16.3 with minimal modifications. Will push shortly.